k9mail/src/main/java/security/planck/planckinvite/PlanckInvitationSender.kt

```diff
@@ -0,0 +1,100 @@
+import javax.inject.Inject
+
+private const val INVITATION_ATTACHMENT_NAME = "email_gradient"
+private const val INVITATION_ATTACHMENT_MIME_TYPE = "image/png"
+private const val INVITATION_PARAGRAPH_1_PLACE_HOLDER = "PARAGRAPH_1"
+private const val INVITATION_PARAGRAPH_2_PLACE_HOLDER = "PARAGRAPH_2"
+private const val INVITATION_BUTTON_TEXT_PLACE_HOLDER = "BUTTON_TEXT"
+private const val INVITATION_PARAGRAPH_3_PLACE_HOLDER = "PARAGRAPH_3"
+
+class PlanckInvitationSender @Inject constructor(
+    @ApplicationContext private val context: Context,
+    private val rawResources: RawResources,
+    private val invitationAttachmentCreator: RawResourceAttachmentCreator,
+    private val messagingRepository: MessagingRepository,
+    private val dispatcherProvider: DispatcherProvider,
+) {
+    private val uiScope = CoroutineScope(dispatcherProvider.main())
+
+
+    fun sendPlanckInvitesToUnsecureRecipients(
+        recipients: List<Address>,
+        sender: Identity,
+        account: Account,
+    ) {
+        uiScope.launch {
+            val messageBuilder = createMessageBuilderToSendPlanckInvites(
+                recipients,
+                sender,
+                account
+            )
+            messagingRepository.buildAndSendMessage(messageBuilder, account)
+        }
+    }
+    private suspend fun createMessageBuilderToSendPlanckInvites(
+        recipients: List<Address>,
+        sender: Identity,
+        account: Account,
+    ): MessageBuilder {
+        val builder: MessageBuilder = SimpleMessageBuilder.newInstance()
+        val invitationTemplate: String = rawResources.readTextFileFromRaw(R.raw.planck_invite)
+        val invitationText = invitationTemplate.replace(
+            INVITATION_PARAGRAPH_1_PLACE_HOLDER,
+            context.getString(R.string.planck_invite_paragraph_1, sender.email)
+        ).replace(
+            INVITATION_PARAGRAPH_2_PLACE_HOLDER,
+            context.getString(R.string.planck_invite_paragraph_2)
+        ).replace(
+            INVITATION_BUTTON_TEXT_PLACE_HOLDER,
+            context.getString(R.string.planck_invite_button_text)
+        ).replace(
+            INVITATION_PARAGRAPH_3_PLACE_HOLDER,
+            context.getString(R.string.planck_invite_paragraph_3, sender.email)
+        )
+        val invitationAttachments: Map<String, Attachment> = createInvitationAttachment()
+
+        return builder.setSubject(context.getString(R.string.planck_invite_title))
+            .setSentDate(Date())
+            .setHideTimeZone(K9.hideTimeZone())
+            .setTo(recipients)
+            .setIdentity(sender)
+            .setMessageFormat(SimpleMessageFormat.HTML)
+            .setText(invitationText)
+            .setInlineAttachments(invitationAttachments)
+            .setSignature(account.signature)
+            .setSignatureBeforeQuotedText(account.isSignatureBeforeQuotedText)
+            .html()
+    }
+
+    private suspend fun createInvitationAttachment(): Map<String, Attachment> {
```

Comment:
what is the attachment for?

Reply:
It is the inline attachment for the inline image that is displayed in the invitation email.